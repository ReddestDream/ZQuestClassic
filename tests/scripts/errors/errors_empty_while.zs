// https://discord.com/channels/876899628556091432/1379668042044145745
npc script Quickknife {
    void run() {
        while (true) {
        }
    }
}
