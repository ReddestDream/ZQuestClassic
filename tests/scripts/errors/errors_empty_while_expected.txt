stderr:

Compiling 'errors_empty_while.zs'
Pass 1: Parsing
Pass 2: Preprocessing
Pass 3: Registration
Pass 4: Analyzing Code
Pass 5: Checking code paths
Pass 6: Generating object code
Pass 7: Assembling
Success!
Compile finished with exit code '0' (success)

stdout:

{
  "success": true,
  "diagnostics": [],
  "metadata": {
    "currentFileSymbols": 2,
    "symbols": 2,
    "identifiers": 2,
    "elided": true
  }
}

zasm:

QUIT; void run() Body
PUSHR REFNPC; void run() Body Start
SETR D4 SP2
GOTO 4; while() #1 End AND [Opt:NeverRet] AND void run() Body End
